mdfreader/mdf3reader.py

```diff
@@ -949,7 +949,12 @@ def read3(self, fileName=None, info=None, multiProc=False, channelList=None, \
 
                         for chan in channels: # for each recordchannel
                             recordName = buf[recordID]['record'].recordToChannelMatching[chan.recAttributeName]  # in case record is used for several channels
-                            temp = buf[recordID]['data'][recordName]
+                            try:
```

Comment:
This particular try I feel is necessary.  For whatever reason when my file is read there are many recordName values that don't exist in buf.  Because of this, the program will crash here.  I noticed an open issue where someone was complaining about the same crash in the same place, I bet it is the same issue.  Another solution would be a routine that checks if recordName exists in buf and if not, pop() is out?  This might be a similar performance hit though.